lib/galaxy/schema/schema.py

```diff
@@ -0,0 +1,359 @@
+
+
+class HDADetailed(HDASummary):
+    model_class: str = ModelClassField(HDA_MODEL_CLASS_NAME)
+    hda_ldda: str = Field(
+        "hda",
+        const=True,
+        title="HDA or LDDA",
+        description="Whether this dataset belongs to a history (HDA) or a library (LDDA).",
+        deprecated=False  # TODO Should this field be deprecated in favor of model_class?
+    )
+    accessible: bool = AccessibleField
+    genome_build: str = Field(
+        "?",
+        title="Genome Build",
+        description="TODO",
+    )
+    misc_info: str = Field(
+        ...,
+        title="Miscellaneous Information",
+        description="TODO",
+    )
+    misc_blurb: str = Field(
+        ...,
+        title="Miscellaneous Blurb",
+        description="TODO",
+    )
+    file_ext: str = Field(  # Is this a duplicate of HDASummary.extension?
+        ...,
+        title="File extension",
+        description="The extension of the file.",
+    )
+    file_size: int = Field(
+        ...,
+        title="File Size",
+        description="The file size in bytes.",
+    )
+    resubmitted: bool = Field(
+        ...,
+        title="Resubmitted",
+        description="Whether the job creating this dataset has been resubmitted.",
+    )
+    metadata: Any = Field(  # TODO: create pydantic model for metadata?
+        ...,
+        title="Metadata",
+        description="The metadata associated with this dataset.",
+    )
+    meta_files: List[MetadataFile] = Field(
+        [],
+        title="Metadata Files",
+        description="Collection of metadata files associated with this dataset.",
+    )
+    data_type: str = Field(
+        ...,
+        title="Data Type",
+        description="The fully qualified name of the class implementing the data type of this dataset.",
+        example="galaxy.datatypes.data.Text"
+    )
+    peek: str = Field(
+        ...,
+        title="Peek",
+        description="A few lines of contents from the start of the file.",
+    )
+    creating_job: str = Field(
+        ...,
+        title="Creating Job ID",
+        description="The encoded ID of the job that created this dataset.",
+    )
+    rerunnable: bool = Field(
+        ...,
+        title="Rerunnable",
+        description="Whether the job creating this dataset can be run again.",
+    )
+    uuid: UUID4 = Field(
+        ...,
+        title="UUID",
+        description="Universal unique identifier for this dataset.",
+    )
+    permissions: DatasetPermissions = Field(
+        ...,
+        title="Permissions",
+        description="Role-based access and manage control permissions for the dataset.",
+    )
+    file_name: FilePath = Field(
+        ...,
+        title="File Name",
+        description="The full path to the dataset file.",
+    )
+    display_apps: List[DisplayApp] = Field(
+        [],
+        title="Display Applications",
+        description="Contains new-style display app urls.",
+    )
+    display_types: List[DisplayApp] = Field(
```

Comment:
I would love to deprecate this AND display_apps AND visualizations... this is dictated by the data type and can be mapped that way on the client.

Reply:
I thought this too but it looks like there is a filter when I was looking over #11880 which filters by dbkey.

Reply:
:( that is unfortunate. We might still be able to do it on the client (and there would be great value in it, this can be the bulk of data transferred in the details view), but that is a good bit more fiddly then.